### PR TITLE
Ci add luacheck

### LINKED ISSUE #1253: Run luacheck in CI
Currently, [luacheck](https://github.com/mpeterv/luacheck) reports 626 warnings:
```
Total: 626 warnings / 0 errors in 96 files
```
Would it be of interest to add luacheck to the CI? If so I'll make a PR for this.

### PR DESCRIPTION
**Pull request by carlsmedstad**
Resolves #1253.

I tried to do this as non-intrusively as possible since I'm not very familiar with the code base. Hopefully, the fixes and ignores are considered appropriate. If not, I'm very open to reworking them with feedback from the maintainers.

### CODE REVIEW

File: src/luarocks/tools/zip.lua [hunk 1]
```diff
@@ -10,7 +10,7 @@ local dir = require("luarocks.dir")
 
 local pack = table.pack or function(...) return { n = select("#", ...), ... } end
 
-local stat_ok, stat = pcall(require, "posix.sys.stat")
+pcall(require, "posix.sys.stat")
```
[the remark this answers is not in the document]
Comment (carlsmedstad) [reply src/luarocks/tools/zip.lua:13]: I assumed so as well, but was not certain, so I left it. I'll remove it and see if it breaks any tests.

**Comment by hishamhm** [comment]
Hi there! Thank you for contributing! :)

I've never integrated luacheck in CI before because I already have it integrated into my editor, so most of the serious things that luacheck catches I tend to spot immediately, and there were a couple of things that luacheck does warn about that I never found important enough (I've recently noticed that I had never merged in my `.luacheckrc` after all these years, so hopefully that gives an idea of the kinds of errors I usually ignore). Plus, luacheck is currently in need of a maintainer, so I've been reluctant on adding it as a dependency.

Still, I see this PR has caught a couple of mistakes here and there, which is nice!

One change I find concerning, though, is the many variables renamed to `_`. Given a dynamically typed code-base, having the proper names serves as useful documentation to what the functions return, even if we're not using currently them right now. I wonder if [this feature](https://luacheck.readthedocs.io/en/stable/warnings.html#secondary-values-and-variables) could silence those warnings and keep those variables properly named?

**Comment by carlsmedstad** [comment]
Hey! :wave:

My pleasure, I'm very appreciative of Lua having such good package management tooling so I'm glad to be able to contribute :) 

Your suggestion of using the option `unused_secondaries` did indeed remove the warnings. I've updated the PR and squashed the commits.

**Comment by hishamhm** [comment]
@carlsmedstad Thank you for the adjustments! It's now merged :)

**Comment by hishamhm** [comment]
I guess I'll add a luacheck step to my personal git pre-commit hook now!

**Comment by hishamhm** [comment]
```
$ cat .git/hooks/pre-commit
#!/bin/sh

luacheck --cache .luacheckcache -q src/
```

**Pull request merged**